build/elasticsearch/bin/es-docker

```diff
@@ -36,4 +35,28 @@ done < <(env)
 # will run in.
 export ES_JAVA_OPTS="-Des.cgroups.hierarchy.override=/ $ES_JAVA_OPTS"
 
+# Determine if x-pack platinum is enabled
+if bin/elasticsearch-plugin list -s | grep x-pack; then
+    # Setting ELASTIC_PWD is mandatory on the *first* node (unless
+    # LDAP is used). As we have no way of knowing if this is the first
+    # node at this step, we can't enforce the presence of this env
+    # var.
+    if [ -n "$ELASTIC_PWD" ]; then
+        [[ -f config/elasticsearch.keystore ]] || bin/elasticsearch-keystore create
+        echo "$ELASTIC_PWD" | bin/elasticsearch-keystore add -x 'bootstrap.password'
+    fi
+
+    # Permit the use of a pre-bundled self signed cert for transport TLS
+    # through ALLOW_INSECURE_DEFAULT_TLS_CERT flag
+    # This should be used strictly for on non production environments.
+    if [ ! -z ${ALLOW_INSECURE_DEFAULT_TLS_CERT+dummy} ]; then
```

Comment:
Nice, rigorous shell work. 👍

Can we standardize on Bash built-in test (`[[`) though?